globus_cli/commands/endpoint/search.py

```diff
@@ -26,6 +26,9 @@ def endpoint_search(filter_fulltext, filter_owner_id, filter_scope):
     """
     Executor for `globus endpoint search`
     """
+    if filter_scope == 'all' and not filter_fulltext:
+        raise click.UsageError('filter_scope "all" requires a search filter.')
```

Comment:
I think this message might still be confusing unless you literally typed in `globus endpoint search --filter-scope=all` and/or are cognizant of the `--filter-scope` option.

Maybe: `When searching all endpoints (--filter-scope=all, the default), a full-text search filter is required. Other scopes (e.g. --filter-scope=recently-used) may be used without specifying an additional filter.`